### ISSUE TITLE
[FEATURE] support large files

### ISSUE THREAD

**Issue by sokoow** [HemmeligOrg/Hemmelig.app#163]
### Description

Support large files - currenctly when I try to upload anything more than > 40 megs, my browser would hang with SIGILL . Is it possible to support files larger than 1GB?

**Comment by bjarneo** [comment]
Did not see this one until now. So, that is a great question. Since the encryption is done by tweetnacl in the browser, it is using browser resources. I have to play around and test ti ser what is possible or not. Find the bottleneck.

**Comment by bjarneo** [comment]
The issue is that the FileReader API reads te files directly into memory, which will crash the browser for large files. To make this work, we have to use the `readAsArrayBuffer()` functionality for large files.

**Comment by bjarneo** [comment]
It requires a lot of refactoring to make this work.

By default everything was base64 encoded (after the encryption to make it easy to post to the server), which works fine for normal text, but not large files

What has to be done is
1. use this API in the zip.js file: reader.readAsArrayBuffer(file); (place it next to new FileReader()
2. Remove encode base64 from zipFiles function
3. Update both encrypt and decrypt in the crypto.js file to not use base64encode or decode for files
4. It has to be handled differently by the upload, currently it is stringified form data added to a POST request in the api/secrets.js#createSecret() function
5. Handle streaming of large files to the server in a fine way

So, TL;DR: to make this work it requires a lot of work, but, should definitely be done at some point.
